### ISSUE TITLE
Panic in binlog player when merging using SplitClone

### ISSUE THREAD

**Issue by dweitzman** [vitessio/vitess#4551]
I attempted a shard merge using SplitClone. 

The vreplication row it inserts into _vt.vreplication set "source" to `keyspace:"foo" shard:"-80"`. The binlog player panics (which is recovered, so no tablet crash) if I don't modify it to be `keyspace:"foo" shard:"-80" key_range:<>`.

E0125 01:31:30.778846   18173 controller.go:143] stream 6: caught panic: runtime error: invalid memory address or nil pointer dereference
/usr/local/go/src/runtime/panic.go:513 (0x42e4e8)
/usr/local/go/src/runtime/panic.go:82 (0x42d63d)
/usr/local/go/src/runtime/signal_unix.go:390 (0x444461)
/vt/src/vitess.io/vitess/go/vt/binlog/binlogplayer/binlog_player.go:236 (0xc033e5)
/vt/src/vitess.io/vitess/go/vt/binlog/binlogplayer/binlog_player.go:177 (0xc01627)
/vt/src/vitess.io/vitess/go/vt/vttablet/tabletmanager/vreplication/controller.go:183 (0xc8a166)
/vt/src/vitess.io/vitess/go/vt/vttablet/tabletmanager/vreplication/controller.go:124 (0xc89b8b)
/usr/local/go/src/runtime/asm_amd64.s:1333 (0x45e730)
W0125 01:31:30.778869   18173 controller.go:134] stream 6: panic: runtime error: invalid memory address or nil pointer dereference, retrying after 5s

Basically, binlog player's `keyRange` field can be a nil pointer but binlog player dereferences it without first verifying that

**Comment by makmanalp** [comment]
#4571 merged, should probably close

**Comment by GuptaManan100** [comment]
Closing the issue. Reopen in case of new developments.
